src/main/kotlin/com/criptext/mail/api/PeerEventsApiHandler.kt

```diff
@@ -0,0 +1,136 @@
+package com.criptext.mail.api
+
+import com.criptext.mail.api.models.Event
+import com.criptext.mail.db.dao.PendingEventDao
+import com.criptext.mail.utils.PeerQueue
+import com.criptext.mail.utils.ServerErrorCodes
+import com.github.kittinunf.result.Result
+import org.json.JSONArray
+import org.json.JSONObject
+
+interface PeerEventsApiHandler {
+    fun readEmail(metadataKeys: List<Long>, unread: Boolean)
+    fun readThread(threadIds: List<String>, unread: Boolean)
+    fun changeEmailLabel(metadataKeys: List<Long>, labelsRemoved: List<String>,
+                         labelsAdded: List<String>)
+    fun changeThreadLabel(threadIds: List<String>, labelsRemoved: List<String>,
+                          labelsAdded: List<String>)
+    fun deleteEmailPermanently(metadataKeys: List<Long>)
+    fun deleteThreadPermanently(threadIds: List<String>)
+    fun createLabel(text: String, color: String)
+    fun openEmail(metadataKeys: List<Long>)
+    fun checkForMorePendingEvents(): Boolean
+    fun dispatchEvents(): Result<Unit, Exception>
```

Comment:
Convert these to classes to make it more extensible.

Or maybe you could create data classes for each event and provide a `toJSON()` method so that this interface only needs one method to enqueue all